code/modules/clothing/glasses/_glasses.dm

```diff
@@ -183,6 +183,12 @@
 	desc = "Yarr."
 	icon_state = "eyepatch"
 	inhand_icon_state = null
+	actions_types = list(/datum/action/item_action/toggle)
+
+/obj/item/clothing/glasses/eyepatch/attack_self(mob/user, modifiers)
+	. = ..()
+	icon_state = icon_state == "eyepatch_flipped" ? "eyepatch" : "eyepatch_flipped"
```

Comment:
this really shouldn't be hardcoded. Use a variable for the flipped and non-flipped states